src/Spice86.Core/Emulator/Devices/Timer/InvalidCounterIndexException.cs

```diff
@@ -1,12 +1,28 @@
 ﻿namespace Spice86.Core.Emulator.Devices.Timer;
 
+using Spice86.Core.Emulator.CPU;
 using Spice86.Core.Emulator.Errors;
-using Spice86.Core.Emulator.VM;
 
 using System;
 
+/// <summary>
+/// The exception thrown when an invalid counter index is used in the Programmable Interval Timer.
+/// </summary>
 [Serializable]
 public class InvalidCounterIndexException : InvalidVMOperationException {
-    public InvalidCounterIndexException(Machine machine, int counterIndex) : base(machine, $"Invalid counter index {counterIndex}") {
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="state">The CPU state.</param>
+    /// <param name="counterIndex">The incorrect counter index that triggered the exception.</param>
+    public InvalidCounterIndexException(State state, int counterIndex) : base(state, $"Invalid counter index {counterIndex}") {
+    }
+
+    /// <inheritdoc />
+    public InvalidCounterIndexException(State state, string message) : base(state, message) {
+    }
+
+    /// <inheritdoc />
+    public InvalidCounterIndexException(State state, Exception e) : base(state, e) {
```

Comment:
Do you need to redefine the constructor 3 times?

Reply:
It's additional ctors suggested by Rider and Visual Studio. It removed a warning.

Reply:
Additional ctors in exceptions compared to master removed.